src/Opaleye/Operators.hs

```diff
@@ -37,6 +37,18 @@ restrict :: QueryArr (Column T.PGBool) ()
 restrict = QueryArr f where
   f (Column predicate, primQ, t0) = ((), PQ.restrict predicate primQ, t0)
 
+{-| Add a @WHERE EXSITS@ clause to the current query. -}
+exists :: QueryArr a b -> QueryArr a ()
+exists criteria = QueryArr f where
+  f (a, primQ, t0) = ((), PQ.exists primQ existsQ, t0) where
+    (_, existsQ, _) = runSimpleQueryArr criteria (a, t0)
```

Comment:
Wouldn't that only matter if we were returning columns from `criteria`? The `EXISTS` entirely removes any bindings introduced by `criteria`.